qadence/mitigations/analog_zne.py

```diff
@@ -7,33 +7,133 @@
 from torch import Tensor
 
 from qadence import BackendName
-from qadence.backend import ConvertedCircuit, ConvertedObservable
 from qadence.backends.api import backend_factory
 from qadence.backends.pulser.backend import Backend
 from qadence.blocks import block_to_tensor
+from qadence.blocks.abstract import AbstractBlock
+from qadence.blocks.analog import ConstantAnalogRotation, WaitBlock
+from qadence.circuit import QuantumCircuit
 from qadence.measurements import Measurements
 from qadence.mitigations import Mitigations
 from qadence.noise import Noise
+from qadence.operations import AnalogRot
+from qadence.transpile import apply_fn_to_blocks
 from qadence.utils import Endianness
 
 
-def zne(noise: Noise, zne_datasets: list) -> list:
+def zne_pulse(
+    stretches: Tensor, zne_datasets: list[list], n_observables: int, n_params: int
```

Comment:
"n_params" makes it look like the count of parameters in the circuit. I think it refers to the number of different circuit configurations you are executing instead, give a different name maybe if so.

Reply:
See above.